### PR TITLE
Fix idgen+auditable

### PR DESCRIPTION
**Pull request by Blackbaud-MikeLueders**
Call markAudited prior to setting the auto-generated id so the audit infrastructure will correctly detect the entity as new on initial save.

### CODE REVIEW

File: sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/AuditableIdGeneratedRepository.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.azure.spring.data.cosmos.repository.repository;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/AuditableIdGeneratedRepository.java:1]: please add new file license.

File: sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/ReactiveAuditableIdGeneratedRepository.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.azure.spring.data.cosmos.repository.repository;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/ReactiveAuditableIdGeneratedRepository.java:1]: new file license missing here.

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
Overall looks good to me, thanks @Blackbaud-MikeLueders for the changes. 
Just a couple of missing license headers.

File: sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/AuditableIdGeneratedRepository.java (hunk 1 above)
Comment (Blackbaud-MikeLueders) [reply sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/AuditableIdGeneratedRepository.java:3]: updated

File: sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/ReactiveAuditableIdGeneratedRepository.java (hunk 2 above)
Comment (Blackbaud-MikeLueders) [reply sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/repository/ReactiveAuditableIdGeneratedRepository.java:3]: updated

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**